### DISCUSSION TITLE
Is it possible to delete an entire instance?

### DISCUSSION THREAD

**Discussion by gabimoncha** [Tencent/MMKV#837] [Q&A] [answered]
Is it possible to delete an entire instance?

**Comment by lingol** [comment]
Just call `close()`. Be careful though, once closed, you should never call any other methods on that instance.

**Comment by lingol** [comment]
Oh, if you want to delete the underlying storage as well, you should call OS's `remove()` or `unlink()` or something like that to delete the file & CRC file directly. MMKV doesn't provide methods to delete the storage.

**Comment by gabimoncha** [comment]
So if I want to completely wipe out the storage and mmkv instance, I would first have to remove keys and then close the instance?

I'm asking because in my react native app, I remove the keys from the encrypted storage, but I cannot initialise another instance with the same id, but different encryption key.

I first have to check if the instance was created before, in this case I empty and decrypt it on removal, then next time I instance it with the same id and encrypt it with a new key right after.

**Comment by gabimoncha** [comment]
I tried looking before in the source code but didn't manage to find those methods, that's why I asked here. Can you please link it for me?

**Comment by lingol** [comment, marked as answer]
In your scenario, you may be looking for `reKey()`. Here's how it's done right with reKey:
1. Open the existing instance with the old encryption key.
2. Delete existing keys by calling `remove()` or `removeAll()`.
3. Call `reKey()` with the new encryption key.
4. Use the instance as if it's a new one.

**Reply by gabimoncha** [reply]
Thank you @lingol! But this still needs me to check if the I created an instance before this.
Can't I just remove it and then create a new one?
From what you've said, calling `clear()` would remove instance with keys all together?
Then next time, I can use the same code for creating an encrypted instance (even if it has the same id, but a different encryption key)

**Reply by lingol** [reply]
No. You can't do that. The file keeps its last encryption info even empty (after `removeAll()` or `clear()`). There are no methods inside MMKV for you to erase the underlying storage. Like I have mentioned before, you have to call the system's method to do it.

**Reply by lingol** [reply]
OS means operating system, by the way.

**Reply by gabimoncha** [reply]
Ok, thank you for your help.

**Comment by lingol** [comment]
`removeStorage()` is added in v1.3.3.
